packages/ses/src/make-hardener.js

```diff
@@ -27,26 +27,80 @@ import {
   TypeError,
   WeakMap,
   WeakSet,
+  apply,
   arrayForEach,
+  defineProperty,
   freeze,
+  getOwnPropertyDescriptor,
   getOwnPropertyDescriptors,
   getPrototypeOf,
+  isInteger,
   isObject,
   objectHasOwnProperty,
   ownKeys,
+  seal,
   setAdd,
   setForEach,
   setHas,
+  toStringTagSymbol,
+  typedArrayPrototype,
   weakmapGet,
   weakmapSet,
   weaksetAdd,
   weaksetHas,
 } from './commons.js';
+import { assert } from './error/assert.js';
 
 /**
  * @typedef {import('../index.js').Harden} Harden
  */
 
+// Obtain the string tag accessor of of TypedArray so we can indirectly use the
+// TypedArray brand check it employs.
+const typedArrayToStringTag = getOwnPropertyDescriptor(
+  typedArrayPrototype,
+  toStringTagSymbol,
+);
+assert(typedArrayToStringTag);
+const getTypedArrayToStringTag = typedArrayToStringTag.get;
+assert(getTypedArrayToStringTag);
+
+// Exported for tests.
+/** @param {unknown} object */
+export const isTypedArray = object => {
+  // The object must pass a brand check or toStringTag will return undefined.
+  const tag = apply(getTypedArrayToStringTag, object, []);
```

Comment:
How bad is the overhead of empty array allocation these days?

Reply:
This is hot enough to care, but it’s not the long pole on this line. `uncurryThis` is in play, and it allocates profusely.

Reply:
(I imagine that when we begin to care earnestly about SES performance, we will be compelled to examine how much care we need to give toward protection from flaws in the TCB like vetted shims, and how much SES can lean on Lockdown to protect its own invariants instead of these performance-expensive measures.)